### PR TITLE
add weather to closet board

### PR DESCRIPTION
**Pull request by happysituations**

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    });
+});
+
+function sendWeather (locationKey, res) {
+  Axios.get(`http://dataservice.accuweather.com/forecasts/v1/daily/1day/${locationKey}?` +
+  `apikey=${ACCUWEATHER_KEY}`).then(function (respsonse) {
+    res.send(respsonse.data);
+  }).catch(function (error) {
+    res.sendStatus(500);
+    console.log(error);
```
Comment (skoshkarli) [inline src/server/index.js]: if we are responding with a 500, we should propogate the error to the UI. no point of console loggin it.

File: src/server/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+var port = process.env.PORT || 3000;
+var express = require('express');
+var path = require('path');
+var Axios = require('axios');
+var app = express();
+// in console, run this command: export ACCUWEATHER_KEY=YOURKEY
```
Comment (hcnimi) [inline src/server/index.js]: Should we include instructions in readme?

File: src/server/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const ACCUWEATHER_KEY = process.env.ACCUWEATHER_KEY;
+
+// e.g. http://localhost:3000/api/locationkey?lat=30.37&lon=-97.76
+app.get('/api/locationkey', (req, res) => {
+  var coordinates = req.query.lat + '%2C' + req.query.lon;
+  console.log('http://dataservice.accuweather.com/locations/v1/cities/geoposition/search?' +
```
Comment (hcnimi) [inline src/server/index.js]: please remove console.log.

File: src/server/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+var port = process.env.PORT || 3000;
```
Comment (skoshkarli) [inline src/server/index.js]: any reason for the rename? we should keep jsx?

File: src/server/index.js (hunk 2 above)
Comment (skoshkarli) [reply src/server/index.js]: +1

File: src/server/index.js (hunk 4 above)
Comment (hcnimi) [reply src/server/index.js]: shouldn't it be .js? it's backend.

File: src/server/index.js (hunk 2 above)
Comment (skoshkarli) [reply src/server/index.js]: ah yeah i guess.

File: src/server/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+app.get('/api/locationkey', (req, res) => {
+  var coordinates = req.query.lat + '%2C' + req.query.lon;
+  console.log('http://dataservice.accuweather.com/locations/v1/cities/geoposition/search?' +
+  `apikey=${ACCUWEATHER_KEY}&q=${coordinates}`)
+  Axios.get('http://dataservice.accuweather.com/locations/v1/cities/geoposition/search?' +
+    `apikey=${ACCUWEATHER_KEY}&q=${coordinates}`).then(function (respsonse) {
+      sendWeather(respsonse.data.Key, res);
+    }).catch(function (error) {
+      res.sendStatus(500);
+      console.log(error);
```
Comment (skoshkarli) [inline src/server/index.js]: if we are responding with a 500, we should propogate the error to the UI. no point of console loggin it.

File: package.json [hunk 6]
```diff
@@ -30,33 +30,35 @@
...
     "babel-preset-env": "^1.7.0",
     "babel-preset-react": "^6.24.1",
+    "chai": "^4.1.2",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-15": "^1.0.5",
     "enzyme-adapter-react-16": "^1.1.1",
     "html-webpack-plugin": "^3.2.0",
     "jest": "^23.2.0",
     "jest-enzyme": "^6.0.2",
+    "n": "^2.1.12",
```
Comment (skoshkarli) [inline package.json]: what's this package for?

Comment (happysituations) [reply package.json]: That was a typo with npm install, I removed it XD

File: src/server/index.js (hunk 1 above)
Comment (happysituations) [reply src/server/index.js]: I updated the API to check for the appropriate input and give a useful error to the user (of the API) for bad input. If we get to the point where the request to AccuWeather gives an error, there’s nothing the user can do about it so it returns a 500 and logs the error on the server. In a real work environment, logging on the server is important for developers to find errors down the line (this is huge at my work, also they say its bad practice to send the whole error for an internal server problem to the user but its okay to send a friendly message instead).

File: src/server/index.js (hunk 2 above)
Comment (happysituations) [reply src/server/index.js]: done

File: src/server/index.js (hunk 4 above)
Comment (happysituations) [reply src/server/index.js]: It should be .js

File: src/client/components/ClosetBoard.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Axios from 'axios';
+
+export class MyCloset extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
[the remark this answers is not in the document]
Comment (happysituations) [reply src/client/components/ClosetBoard.jsx]: Do you mind walking me through how to refactor this to use redux?

**Pull request merged**